Integrate iitkgp-erp-login package and resolve faculty department dat…

Faizan2005 opened the pull request.

Faizan2005 committed:
Integrate iitkgp-erp-login package and resolve faculty department data fetching issue

Faizan2005 commented:
@harshkhandeparkar Please review the PR

harshkhandeparkar commented on .session:
bruh :)

harshkhandeparkar replied:
Add this file to `.gitignore`

harshkhandeparkar commented on main.py:
```diff
@@ -15,41 +12,121 @@
+        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        'Cookie': 'ci_session=gk7bi3u0r2ej0h3qnihk4gocf0vp9e8e',
+        'Host': 'www.iitkgp.ac.in',
+        'Origin': 'https://www.iitkgp.ac.in',
```
Are all of these headers required?

Faizan2005 replied:
Initially I had added some selected headers but it wasn't solving the problem, the data wasn't getting scraped so I copied all the headers, then it started working fine. So, didn't do anything much with that.

harshkhandeparkar commented on main.py:
```diff
@@ -15,41 +12,121 @@
+            more_pages = False
+        else:
+            for prof in dept_raw_data:
+                name = re.findall(r">(.+?)<", prof["faculty"])[0].replace("  ", "")
```
Can you add some comments explaining the code?

Faizan2005 replied:
Okay, I'll add comments and update the code.

harshkhandeparkar commented on main.py:
```diff
@@ -64,57 +141,22 @@ def parse_html(dep):
-    """
-    Note:
-
-    If a prof teaches subjects from other departments,
```
I think this note is quite helpful, can you add it back? It isn't irrelevant anymore, is it?

harshkhandeparkar replied:
@Faizan2005

harshkhandeparkar commented on main.py:
```diff
@@ -124,61 +166,46 @@ def parse_html(dep):
             result = list(result for result, _ in itertools.groupby(result))
-
-    except:
+    except KeyError:
```
Could KeyError be thrown here?

harshkhandeparkar commented on main.py:
```diff
@@ -188,53 +215,35 @@ def get_table(details):
-        sys.exit(1)
-
-    cookie = {"JSESSIONID": os.getenv("JSESSIONID")}
+    headers = {
```
These headers seem to be repeating. Perhaps make it a constant somewhere and import?

Faizan2005 replied:
Okay, I'll make the changes

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to .

Faizan2005 committed:
Added .session to .gitignore

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to config.pyto:              cannot open `to' (No such file or directory)
store:           cannot open `store' (No such file or directory)
headers:         cannot open `headers' (No such file or directory)
and:             cannot open `and' (No such file or directory)
payload:         cannot open `payload' (No such file or directory)
configurations.: cannot open `configurations.' (No such file or directory).sessionto: cannot open `to' (No such file or directory).gitignore'.

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to .

Faizan2005 commented:
@harshkhandeparkar I have made the changes. 
1. there was some issue in parsing the prof data, this has been fixed. 
2. moved the headers and payload to a separate config.py file. 
3. functions like get_times(), get_table() etc which were being used by app.py have been moved to app.py.
4. added some comments.
5. moved .session file to .gitignore

harshkhandeparkar commented on .session:
Can you please remove this file :slightly_smiling_face: 
![image](https://github.com/user-attachments/assets/ec887be6-07e5-465a-b1ba-665adbb994c7)

harshkhandeparkar commented on ter:
??

Faizan2005 replied:
I think it got created while making commit.
I wanted to make changes to the commit msg, I think during that time it happened. Now has been removed

Faizan2005 committed:
Remove unnecessary files, add comments

- Removed 'ter' file
- Added comments to
- Removed  file

Faizan2005 commented:
@harshkhandeparkar I have made the changes
1. Added the comments
2. Removed .session
3. Removed ter

harshkhandeparkar commented:
Changes included into and superseded by #82
